diff --git a/lib/context.js b/lib/context.js
index c75e9ad5e496dc2ceb924d90a6a3f1fcde321343..3627ae8e2bc0ba580750446a976befbcdf258d8c 100644
--- a/lib/context.js
+++ b/lib/context.js
@@ -61,6 +61,7 @@ export class Context {
         });
         const ext = PathExt.extname(this._path);
         this.sessionContext = new SessionContext({
+            driveName: manager.contents.driveName(this._path),
             kernelManager: manager.kernels,
             sessionManager: manager.sessions,
             specsManager: manager.kernelspecs,
